src/test/java/org/apache/mesos/offer/ResourceCleanerTest.java

```diff
@@ -10,138 +11,266 @@
+            123, 456, ROLE, PRINCIPAL, EXPECTED_RESOURCE_1_ID);
+    private static final Resource EXPECTED_RESOURCE_2 = ResourceBuilder.volume(
+            999.0, ROLE, PRINCIPAL, "expected-path", EXPECTED_RESOURCE_2_ID);
+
+    private static final Resource UNEXPECTED_RESOURCE_1 = ResourceBuilder.volume(
+            1000.0, ROLE, PRINCIPAL, "unexpected-path-1", UNEXPECTED_RESOURCE_1_ID);
+    private static final Resource UNEXPECTED_RESOURCE_2 = ResourceBuilder.reservedCpus(
+            1.0, ROLE, PRINCIPAL, UNEXPECTED_RESOURCE_2_ID);
+    private static final Resource UNEXPECTED_RESOURCE_3 = ResourceBuilder.volume(
+            1001.0, ROLE, PRINCIPAL, "unexpected-path-3", UNEXPECTED_RESOURCE_3_ID);
+
+    private static final TaskInfo TASK_INFO_1 = TaskInfo.newBuilder()
+            .setName("task-name-1")
+            .setTaskId(TaskID.newBuilder().setValue("task-id-1"))
+            .setSlaveId(SlaveID.newBuilder().setValue(SLAVE_ID))
+            .setExecutor(ExecutorInfo.newBuilder()
+                    .setExecutorId(ExecutorID.newBuilder().setValue("hey"))
+                    .setCommand(CommandInfo.newBuilder().build())
+                    .addResources(EXPECTED_RESOURCE_1)
+                    .build())
+            .build();
+    private static final TaskInfo TASK_INFO_2 = TaskInfo.newBuilder()
+            .setName("task-name-2")
+            .setTaskId(TaskID.newBuilder().setValue("task-id-2"))
+            .setSlaveId(SlaveID.newBuilder().setValue(SLAVE_ID))
+            .addResources(EXPECTED_RESOURCE_2).build();
+
+    private final List<ResourceCleaner> emptyCleaners = new ArrayList<>();
+    private final List<ResourceCleaner> populatedCleaners = new ArrayList<>();
+    private final List<ResourceCleaner> allCleaners = new ArrayList<>();
+
+    public ResourceCleanerTest() {
+        // Validate ResourceCleaner statelessness by only initializing them once
+
+        StateStore mockStateStore = mock(StateStore.class);
+
+        // cleaners without any expected resources
+        when(mockStateStore.fetchExecutorNames()).thenReturn(new ArrayList<>());
+        emptyCleaners.add(new ResourceCleaner(Collections.emptyList()));
+        emptyCleaners.add(new ResourceCleaner(mockStateStore));
+
+        // cleaners with expected resources
+        populatedCleaners.add(new ResourceCleaner(Arrays.asList(
+                TASK_INFO_1.getExecutor().getResources(0), TASK_INFO_2.getResources(0))));
+        when(mockStateStore.fetchExecutorNames())
+                .thenReturn(Arrays.asList("a", "b"));
+        when(mockStateStore.fetchTasks("a"))
+                .thenReturn(Arrays.asList(TASK_INFO_1));
+        when(mockStateStore.fetchTasks("b"))
+                .thenReturn(Arrays.asList(TASK_INFO_2));
+        populatedCleaners.add(new ResourceCleaner(mockStateStore));
+
+        allCleaners.addAll(emptyCleaners);
+        allCleaners.addAll(populatedCleaners);
+    }
+
+    @Test
+    public void testNoOffers() {
```

Comment:
Could you add another test which generates no recommendations when presented with All expected resources and volumes?